docs(calendar): improve calendar documentation (#1873)
IgniteUI/igniteui-docfx: en/components/calendar.md

Let's go ahead and try those along with other customizations from the `IgxCalendarComponent API`. First thing we need to set is the `weekStart`, which controls the starting day of the week. It defaults to 0, which corresponds to Sunday, so we will set a value of 1 for Monday. In the markup below we are also binding the `formatOptions` and `formatViews` properties to customize the display formatting. Finally, we are binding the `locale` property to a value, based on the user's location choice:

```html
<!-- app.component.html -->
<igx-calendar #calendar
    [weekStart]="1"
    [locale]="locale"
    [formatOptions]="formatOptions"
    [formatViews]="formatViews">
</igx-calendar>

<igx-select #select [(ngModel)]="locale">
    <igx-select-item *ngFor="let locale of locales" [value]="locale">
        {{ locale }}
    </igx-select-item>

> </igx-select>

Reviewer: That's great, however, I don't see it anywhere in the demo.

</div>

#### Special dates
The [`specialDates`]({environment:angularApiUrl}/classes/igxcalendarcomponent.html#specialdates) feature is using almost the same configuration principles as the `disabledDates`. The ability to select and focus `specialDates` is what differs them from the `disabled` ones.

> Lets add some `specialDates` to our `igxCalendar`. In order to do this, we have to create a [`DateRangeDescriptor`]({environment:angularApiUrl}/interfaces/daterangedescriptor.html) item of type [`DateRangeType.Specific`]({environment:angularApiUrl}/enums/daterangetype.html#specific) and pass an array of dates as a [`dateRange`]({environment:angularApiUrl}/interfaces/daterangedescriptor.html#daterange):

Reviewer: ```suggestion
Let's add some `specialDates` to our calendar. In order to do this, we have to create a [`DateRangeDescriptor`]({environment:angularApiUrl}/interfaces/daterangedescriptor.html) item of type [`DateRangeType.Specific`]({environment:angularApiUrl}/enums/daterangetype.html#specific) and pass an array of dates as a [`dateRange`]({environment:angularApiUrl}/interfaces/daterangedescriptor.html#daterange):
```

```typescript
export class CalendarSample7Component {

>     @ViewChild("calendar", { static: true }) public calendar: IgxCalendarComponent;

Reviewer: ```suggestion
    @ViewChild("calendar", { static: true })
    public calendar: IgxCalendarComponent;
```

>     @ViewChild("alert", { static: true }) public dialog: IgxDialogComponent;

Reviewer: ```suggestion
    @ViewChild("alert", { static: true })
    public dialog: IgxDialogComponent;
```

@import '~igniteui-angular/lib/core/styles/themes/index';
``` 

Following the simplest approach, we create a new theme that extends the [`igx-calendar-theme`]({environment:sassApiUrl}/index.html#function-igx-calendar-theme) and accepts some of the default theme's parameters.

```scss
$custom-calendar-theme: igx-calendar-theme(
  $header-background: #345779,
  $content-background: #fdfdfd,
  $header-text-color: #ffffff,
  $date-current-text-color: #2dabe8,
  $picker-arrow-color: #2dabe8,
  $date-selected-text-color: #fdfdfd,
  $date-current-bg-color: #fdfdfd,
  $picker-arrow-hover-color:  #345779,
  $year-current-text-color: #2dabe8,
  $year-hover-text-color: #2dabe8,
  $month-current-text-color: #2dabe8,
  $month-hover-text-color: #2dabe8,
  $picker-text-color: #2dabe8,
  $picker-text-hover-color:  #345779
);
```

#### Using CSS variables 

The last step is to pass the custom calendar theme: 

```scss
 @include igx-css-vars($custom-calendar-theme);
```

> #### Using mixins

Reviewer: ```suggestion
#### Using Theme Overrides
```

In order to style components for older browsers, like Internet Explorer 11, we have to use a different approach, since it doesn't support CSS variables. 

If the component is using the [`Emulated`](themes/component-themes.md#view-encapsulation) ViewEncapsulation, it is necessary to `penetrate` this encapsulation using `::ng-deep`. To prevent the custom theme to leak into other components, be sure to include the `:host` selector before `::ng-deep`:

 ```scss
:host {
  ::ng-deep {
    @include igx-calendar($custom-calendar-theme);
  }
}
```

> #### Using color palettes

Reviewer: This entire section is not needed. Delete it, please.

  $date-current-text-color: igx-color($custom-calendar-palette, "primary", 50),
  $date-selected-text-color: igx-color($custom-calendar-palette, "secondary", 50),
  $date-current-bg-color: igx-color($custom-calendar-palette, "secondary", 50),
  $picker-arrow-color: igx-color($custom-calendar-palette, "primary", 50),
  $picker-arrow-hover-color: igx-color($custom-calendar-palette, "primary", 500),
  $year-current-text-color: igx-color($custom-calendar-palette, "primary", 50),
  $year-hover-text-color: igx-color($custom-calendar-palette, "primary", 50),
  $month-current-text-color: igx-color($custom-calendar-palette, "primary", 50),
  $month-hover-text-color: igx-color($custom-calendar-palette, "primary", 50),
  $picker-text-color: igx-color($custom-calendar-palette, "primary", 50),
  $picker-text-hover-color: igx-color($custom-calendar-palette, "primary", 500),
  $date-selected-text-color: igx-color($custom-calendar-palette, "secondary", 500)
);
```

>[!NOTE]
>The `igx-color` and `igx-palette` are powerful functions for generating and retrieving colors. Please refer to the [`Palettes`](themes/palette.md) topic for detailed guidance on how to use them.

> #### Using schemas

Reviewer: Same as above. Not needed, delete.
